editor/editor_node.cpp

```diff
@@ -1014,6 +1014,11 @@ void EditorNode::_resources_reimported(const Vector<String> &p_resources) {
 	}
 
 	scene_tabs->set_current_tab(current_tab);
+
+	if (run_surface_upgrade_tool) {
+		SurfaceUpgradeTool::finish_upgrade();
+		run_surface_upgrade_tool = false;
+	}
```

Comment:
Placing it here means that we won't convert internal meshes if there is nothing to reimport. I think you could hook into `_begin_first_scan()` and do the internal mesh conversion before reimporting external scenes and meshes (before triggering the scan).

Reply:
I'm trying this out, and it doesn't seem to work because of the whole editorprogress mess :/

Reply:
Update: I have sorted out this part, but still need to find a solution for the second problem below. Once done, I'll send a diff.